ui/pages/confirmations/components/confirm-page-container/confirm-page-container.component.js

```diff
@@ -213,7 +214,12 @@ const ConfirmPageContainer = (props) => {
           />
         ) : (
           <ConfirmPageContainerHeader
-            showEdit={showEdit}
+            showEdit={
+              origin === ORIGIN_METAMASK ||
+              currentTransaction.type === TransactionType.simpleSend
+                ? showEdit
```

Comment:
We should follow up in a later PR to simplify this logic a bit (I think we can inline the remaining useful parts of `showEdit` here). There might be more dead code to remove as well.

Reply:
We should also follow up with e2e test coverage for any impacted types of transactions we can think of